cpp/include/raft/util/bitset.cuh

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/device_mdarray.hpp>
+#include <raft/core/resource/thrust_policy.hpp>
+#include <raft/core/resources.hpp>
+#include <raft/linalg/map.cuh>
+#include <thrust/for_each.h>
+
+namespace raft::utils {
+/**
+ * @defgroup bitset Bitset
+ * @{
+ */
+/**
+ * @brief View of a RAFT Bitset.
+ *
+ * This lightweight structure stores a pointer to a bitset in device memory with it's length.
+ * It provides a test() device function to check if a given index is set in the bitset.
+ *
+ * @tparam IdxT Indexing type used. Default is uint32_t.
+ */
+template <typename IdxT = uint32_t>
+struct bitset_view {
+  using BitsetT            = uint32_t;
```

Comment:
Any reason not to make this a templated type with the indicated default?

Reply:
Also a nitpick: if we're to keep this alias, could you please consult with clang-tidy on the naming? I think it should be the snake case:
https://github.com/rapidsai/raft/blob/f691fc9651dc83a2e6b3ce7a8de524f4c2a57c44/cpp/.clang-tidy#L205-L206

Reply:
@wphicks Are there use cases for having a different bitset type?

Reply:
For small bitsets, that allows us to avoid wasting memory. In FIL, for instance, we use chars for some of our bitsets and larger data types for others in order to try to back the bitset with a single element when possible but not to use extra memory if we can't. That allows us to perform very efficient bit twiddling operations for the single-element case but only copy exactly as much memory as necessary if we are stuck using multiple elements.